CRYOSMSApp/src/CRYOSMSDriver.cpp

```diff
@@ -869,14 +1206,48 @@ extern "C"
+	static const iocshArg initArg14 = { "useSwitch", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg15 = { "switchTempPv", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg16 = { "switchHigh", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg17 = { "switchLow", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg18 = { "switchStableNumber", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg19 = { "heaterTolerance", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg20 = { "switchTolerance", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg21 = { "heaterOut", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg22 = { "useMagnetTemp", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg23 = { "magnetTempPv", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg24 = { "maxMagnetTemp", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg25 = { "minMagnetTemp", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg26 = { "compOffAct", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg27 = { "noOfComp", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg28 = { "minNoOfComp", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg29 = { "comp1StatPv", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg30 = { "comp2StatPv", iocshArgString };		///< PV Prefix for device
+	static const iocshArg initArg31 = { "rampFile", iocshArgString };		///< PV Prefix for device
+
+	static const iocshArg * const initArgs[] = { &initArg0, &initArg1, &initArg2, &initArg3, &initArg4, &initArg5, &initArg6, &initArg7, &initArg8, &initArg9, &initArg10, &initArg11, 
+		&initArg12, &initArg13, &initArg14, &initArg15, &initArg16, &initArg17, &initArg18, &initArg19, &initArg20, &initArg21, &initArg22, &initArg23, &initArg24, &initArg25, &initArg26,
+		&initArg27, &initArg28, &initArg29, &initArg30, &initArg31 };
 
 	static const iocshFuncDef initFuncDef = { "CRYOSMSConfigure", sizeof(initArgs) / sizeof(iocshArg*), initArgs };
 
 	static void initCallFunc(const iocshArgBuf *args)
 	{
-		CRYOSMSConfigure(args[0].sval, args[1].sval);
+		CRYOSMSConfigure(args[0].sval, args[1].sval, args[2].sval, args[3].sval, args[4].sval, args[5].sval, args[6].sval, args[7].sval, args[8].sval, args[9].sval, args[10].sval, args[11].sval,
+			args[12].sval, args[13].sval, args[14].sval, args[15].sval, args[16].sval, args[17].sval, args[18].sval, args[19].sval, args[20].sval, args[21].sval, args[22].sval, args[23].sval,
+			args[24].sval, args[25].sval, args[26].sval, args[27].sval, args[28].sval, args[29].sval, args[30].sval, args[31].sval);
 	}
```

Comment:
Threre are a lot of arguments, are they always required or are their sensible defaults? If many could be considered optional then using syntax like   ("name1", "value1", "name2", "value2") and  `iocshArgArgv` is another option